src/components/Proposals/ProposalPage/VotingTimelineChart/VotingTimelineChart.tsx

```diff
@@ -0,0 +1,223 @@
+"use client";
+import Image from "next/image";
+import { useState } from "react";
+import {
+  LineChart,
+  Line,
+  XAxis,
+  YAxis,
+  CartesianGrid,
+  ResponsiveContainer,
+  ReferenceLine,
+} from "recharts";
+import { format } from "date-fns";
+import { icons } from "@/icons/icons";
+import { Proposal } from "@/app/api/common/proposals/proposal";
+import { Vote } from "@/app/api/common/votes/vote";
+import Tenant from "@/lib/tenant/tenant";
+import { formatNumber } from "@/lib/utils";
+const { token } = Tenant.current();
+
+function formatNumberWithScientificNotation(x: any) {
```

Comment:
At the first glance this seems unnecessary but I'd like to come back to this method later.